src/java.desktop/windows/native/libawt/windows/ShellFolder2.cpp

```diff
@@ -1088,10 +1089,12 @@ JNIEXPORT jintArray JNICALL Java_sun_awt_shell_Win32ShellFolder2_getIconBits
             // Create java array
             iconBits = env->NewIntArray(nBits);
             if (!(env->ExceptionCheck())) {
-            // Copy values to java array
-            env->SetIntArrayRegion(iconBits, 0, nBits, colorBits);
+                // Copy values to java array
+                env->SetIntArrayRegion(iconBits, 0, nBits, colorBits);
+            }
             free(colorBits);
-        }
+
+            CATCH_BAD_ALLOC_RET(NULL);
```

Comment:
I believe we leak `dc` as well as `iconInfo.hbmColor` and `iconInfo.hbmMask` if `std::bad_alloc` is thrown.

Reply:
Fixed. Now we will release these resources on exit in case of bad_alloc.